src/compile/data/bin.ts

```diff
@@ -24,7 +24,7 @@ function rangeFormula(model: ModelWithField, fieldDef: FieldDef<string>, channel
 
       return {
         formulaAs: field(fieldDef, {binSuffix: 'range'}),
-        formula: `${numberFormatExpr(startField, guide.format, config)} + " - " + ${numberFormatExpr(endField, guide.format, config)}`
+        formula: `${startField} === null ? 'null' : ${numberFormatExpr(startField, guide.format, config)} + " - " + ${numberFormatExpr(endField, guide.format, config)}`
```

Comment:
What if NaN and undefined?   Better use the same expression we use for filtering invalid value?